tmd/Tree/methods.py

```diff
@@ -178,21 +189,36 @@ def get_point_path_distances(self):
     path_lengths = np.append(0, copy.deepcopy(seg_len))
     children = get_children(self)
 
-    for i in children.keys():
-        path_lengths[children[i]] = path_lengths[children[i]] + path_lengths[i]
+    for i, child in children.items():
+        path_lengths[child] = path_lengths[child] + path_lengths[i]
 
     return path_lengths
 
 
+def get_point_trunk_length(self):
+    '''Tree method to get the trunk (first section length).
+    '''
+    ways, end = self.get_sections_only_points()
+    first_section_id = np.where(ways == 0)
+    first_section_start = ways[first_section_id]
+    first_section_end = end[first_section_id]
+    seg_nums = range(first_section_start[0] + 1, first_section_end[0] + 1)
+
+    seg_lengths = get_segment_lengths(self, seg_nums)
+    trunk_length = np.sum(seg_lengths)
+
+    return np.repeat(trunk_length, size(self))
+
+
 def get_point_section_lengths(self):
     '''Tree method to get section lengths.
     '''
     lengths = np.zeros(size(self), dtype=float)
     ways, end = self.get_sections_only_points()
```

Comment:
This does not provide the slices corresponding to the section points.
I think what you want to do is:

```python
begs, ends = self.get_sections_2()

for beg, end in zip(begs, ends):
    lengths[end] = seg_len[beg: end].sum()
```

Reply:
@lidakanari can you verify this please?

Reply:
No we want the `get_sections_only_points` because the seg_len are aligned with the end of each section, not the start.